PWGJE/Tasks/jetvalidationqa.cxx

```diff
@@ -0,0 +1,191 @@
+    mHistManager.add("leadJetConstEta", "leading jet constituent #eta ; #eta ", HistType::kTH1F, {{nBinsEta, -0.9, 0.9}});
+    // leading selected tracks per collision
+    mHistManager.add("leadTrackPt", "leading selected track Pt ; p_{T} (GeV/#it{c})", HistType::kTH1F, {{nBinsPt, 0, 100}});
+    mHistManager.add("leadTrackPhi", "leading selected track #phi ; #phi ", HistType::kTH1F, {{nBinsPhi, 0, 6.4}});
+    mHistManager.add("leadTrackEta", "leading selected track #eta ; #eta ", HistType::kTH1F, {{nBinsEta, -0.9, 0.9}});
+
+    // 2)Jetvalidation from MC
+    mHistManager.add("MCcollisionVtxZ", "MC control collsion VtxZ ; z [cm]", HistType::kTH1F, {{nBins, -15, 15}});
+    // process jet qa
+    mHistManager.add("MCjetPt", "MC inclusive jetPt ; p_{T} (GeV/#it{c})", HistType::kTH1F, {{nBinsPt, 0, 100}});
+    mHistManager.add("MCjetPhi", "MC inclusive jet #phi ; #phi ", HistType::kTH1F, {{nBinsPhi, -3.2, 6.4}});
+    mHistManager.add("MCjetEta", "MC inclusive jet #eta ; #eta ", HistType::kTH1F, {{nBinsEta, -0.9, 0.9}});
+    // process jet constituent qa - constituents as tracks
+    mHistManager.add("MCjetConstTrackPt", "MC inclusive jet constituent Pt ; p_{T} (GeV/#it{c})", HistType::kTH1F, {{nBinsPt, 0, 100}});
+    mHistManager.add("MCjetConstTrackPhi", "MC inclusive jet constituent #phi ; #phi ", HistType::kTH1F, {{nBinsPhi, 0, 6.4}});
+    mHistManager.add("MCjetConstTrackEta", "MC inclusive jet constituent #eta ; #eta ", HistType::kTH1F, {{nBinsEta, -0.9, 0.9}});
+    // cross check the cuts from Run2Hybrid selection
+    mHistManager.add("MCselectedTrackPt", "MC hybrid track Pt ; p_{T} (GeV/#it{c})", HistType::kTH1F, {{nBinsPt, 0, 100}});
+    mHistManager.add("MCselectedTrackPhi", "MC hybrid track #phi ; #phi ", HistType::kTH1F, {{nBinsPhi, 0, 6.4}});
+    mHistManager.add("MCselectedTrackEta", "MC hybrid track #eta ; #eta ", HistType::kTH1F, {{nBinsEta, -0.9, 0.9}});
+  }
+  // add another process for MC studies !
+  void processData(soa::Join<aod::Collisions, aod::EvSels>::iterator const& collision, soa::Join<aod::Jets, aod::JetConstituents> const& jets, TracksJE const& tracks)
+  {
+    if (collision.posZ() > 10) {
+      return;
+    }
+    mHistManager.fill(HIST("collisionVtxZ"), collision.posZ());
+
+    double leadingJetPt = -1;
+    double leadingJetPhi = -1;
+    double leadingJetEta = -1;
+    // jet QA hists per jet in this collision
+    for (const auto& j : jets) {
+      mHistManager.fill(HIST("jetPt"), j.pt());
+      mHistManager.fill(HIST("jetPhi"), j.phi());
+      mHistManager.fill(HIST("jetEta"), j.eta());
+      if (j.pt() > leadingJetPt) {
+        leadingJetPt = j.pt();
+        leadingJetPhi = j.phi();
+        leadingJetEta = j.eta();
+      }
+
+      double leadingConstTrackPt = -1;
+      double leadingConstTrackPhi = -1;
+      double leadingConstTrackEta = -1;
+      // access jet constituents as tracks
+      for (auto& jct : j.tracks_as<TracksJE>()) {
+        if (!validationTracks.IsSelected(jct)) {
```

Comment:
I am not sure about this if statement. The track is already clustered into the jet so removing it here wont change any jet proprties. Of course one can check if the jet finder really is making the correct cut but if you want to find the pt, eta and phi distributions of the jet constituents then this rejection would be out of place